### ISSUE TITLE
terraform 0.13.0-beta2: `terraform -version` doesn't work

### ISSUE THREAD

**Issue by jeremykatz** [hashicorp/terraform#25305]
This is occurring with the Darwin build of Terraform 0.13.0-beta2 running macOS 10.15.5.

```
$ shasum -a 256 terraform_0.13.0-beta2_darwin_amd64.zip 
4729fa267c5be7f1d0c19a85d36e2b4577f303866fc25403650acb4243c2021f  terraform_0.13.0-beta2_darwin_amd64.zip
$ unzip terraform_0.13.0-beta2_darwin_amd64.zip 
Archive:  terraform_0.13.0-beta2_darwin_amd64.zip
  inflating: terraform               
$ TF_LOG=trace ./terraform -version
2020/06/18 20:55:59 [INFO] Terraform version: 0.13.0 beta2 
2020/06/18 20:55:59 [INFO] Go runtime version: go1.14.2
2020/06/18 20:55:59 [INFO] CLI args: []string{"/tmp/terraform", "-version"}
2020/06/18 20:55:59 [DEBUG] Attempting to open CLI config file: /Users/user/.terraformrc
2020/06/18 20:55:59 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2020/06/18 20:55:59 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2020/06/18 20:55:59 [DEBUG] ignoring non-existing provider search directory /Users/user/.terraform.d/plugins
2020/06/18 20:55:59 [DEBUG] ignoring non-existing provider search directory /Users/user/Library/Application Support/io.terraform/plugins
2020/06/18 20:55:59 [DEBUG] ignoring non-existing provider search directory /Library/Application Support/io.terraform/plugins
2020/06/18 20:55:59 [INFO] CLI command args: []string{"version", "-version"}
Error parsing command-line flags: flag provided but not defined: -version

```

**Comment by jeremykatz** [comment]
`terraform -v` results in the same behavior. Something expected to be invalid results in the program usage output
```
$ ./terraform -imadethisup
Usage: terraform [-version] [-help] <command> [args]

The available commands for execution are listed below.
The most common, useful commands are shown first, followed by
less common or more advanced commands. If you're just getting
started with Terraform, stick with the common commands. For the
other commands, please read the help and docs before usage.
[...]
```

**Comment by antonbabenko** [comment]
Additionally, `--version` does not work for in Terraform 0.13 beta 2, but terragrunt and some other custom scripts may rely on this functionality.

**Comment by jeremykatz** [comment]
Are double dash options officially supported? While I've noticed that they do appear to work, the usage only mentions single dashes.

**Comment by jbergknoff-rival** [comment]
FYI this was fixed in https://github.com/hashicorp/terraform/pull/25277 but hasn't seen an official release yet.

**Comment by apparentlymart** [comment]
Thanks for reporting this, @jeremykatz!

Indeed, this is a duplicate of #25301 which was fixed in #25277. The fix will be included in the next prerelease (planned for next week), or you can try building Terraform from source code yourself it you'd like to try it now.

Regarding the double-dash options: the command line options parser Terraform uses treats `-foo` and `--foo` as equivalent. The single-dash form is the convention used in all of the documentation and usage information, but double-dash is a non-canonical alternative that's included for those who are accustomed to it from software that uses GNU-style long options.
